src/validation/data/package.rs

```diff
@@ -61,13 +61,23 @@ impl Package {
     /// Try to create a new one from a [`FileDescriptorProto`].
     pub fn new(descriptor: &FileDescriptorProto) -> Result<Self, PackageError> {
         let mut package = Self {
-            file: descriptor.name().into(),
+            files: vec![descriptor.name().into()],
             name: descriptor.package().to_string(),
             entities: Default::default(),
         };
 
+        package.parse(descriptor)?;
+
+        Ok(package)
+    }
+
+    pub fn add(&mut self, descriptor: &FileDescriptorProto) {
```

Comment:
@xfbs This works but it doesn't feel right... Would be happy to get some opinion here from someone more rusty than me...